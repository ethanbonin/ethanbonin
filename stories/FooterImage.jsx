import React from "react"
import PropTypes from "prop-types";
import {OverlayTrigger, Tooltip} from "react-bootstrap";


export const FooterImage = ({src, alt, label, size}) => {
    return (
        <OverlayTrigger
            placement="top"
            overlay={
                <Tooltip id="button-tooltip-2">{label}</Tooltip>
            }
        >
            {/* eslint-disable-next-line @next/next/no-img-element */}
            <img src={src} alt={alt} style={{width: size, height: size, objectFit: "contain"}} />
        </OverlayTrigger>
    )
}

FooterImage.propTypes = {
    /**
     * The `src` for the image
     */
    src: PropTypes.string.isRequired,
    /**
     * The `alt` for the image
     */
    alt: PropTypes.string.isRequired,
    /**
     * The description when they hover over the image
     */
    label: PropTypes.string.isRequired,

    /**
     * This is the height and width of the image that needs to be passed through,
     * because it's `object-fit: contain`, we only need to pass in one pixel size. If we do that, it will
     * automatically fit to itself
     */
    size: PropTypes.string.isRequired,
}

FooterImage.defaultProps = {
    src: 'https://seeklogo.com/images/C/contentful-logo-C395C545BF-seeklogo.com.png',
    alt: "next Logo",
    label: "This is a sample Footer Image",
    size: "50px",
}
