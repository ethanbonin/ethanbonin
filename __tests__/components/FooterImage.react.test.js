// Link.react.test.js
import React from "react"
import { shallow } from "enzyme"
import {FooterImage} from "@stories/FooterImage";
import toJson from "enzyme-to-json";

describe("FooterImage", () => {
    it("Ensure FooterImage changes after mouse hover", async () => {
        const component = shallow(
            <FooterImage
                src={'https://seeklogo.com/images/C/contentful-logo-C395C545BF-seeklogo.com.png'}
                alt={"next Logo"}
                label={"This is a sample Footer Image"}
                size={"50px"}
            />,
        )

        // Ensure overlay message will be equal to the label passed in
        expect(component.props().overlay.props.children).toBe("This is a sample Footer Image")

        let tree = toJson(component)
        expect(tree).toMatchSnapshot()
        // manually trigger the callback
        component.simulate("mouseover")

        // re-rendering
        tree = toJson(component)
        expect(tree).toMatchSnapshot()

        // manually trigger the callback
        component.simulate("mouseleave")

        // re-rendering
        tree = toJson(component)
        expect(tree).toMatchSnapshot()
    })
})