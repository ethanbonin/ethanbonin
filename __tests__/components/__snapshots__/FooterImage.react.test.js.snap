// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterImage Ensure FooterImage changes after mouse hover 1`] = `
<OverlayTrigger
  defaultShow={false}
  overlay={
    <Tooltip
      id="button-tooltip-2"
      placement="right"
    >
      This is a sample Footer Image
    </Tooltip>
  }
  placement="top"
  trigger={
    Array [
      "hover",
      "focus",
    ]
  }
>
  <img
    alt="next Logo"
    src="https://seeklogo.com/images/C/contentful-logo-C395C545BF-seeklogo.com.png"
    style={
      Object {
        "height": "50px",
        "objectFit": "contain",
        "width": "50px",
      }
    }
  />
</OverlayTrigger>
`;

exports[`FooterImage Ensure FooterImage changes after mouse hover 2`] = `
<OverlayTrigger
  defaultShow={false}
  overlay={
    <Tooltip
      id="button-tooltip-2"
      placement="right"
    >
      This is a sample Footer Image
    </Tooltip>
  }
  placement="top"
  trigger={
    Array [
      "hover",
      "focus",
    ]
  }
>
  <img
    alt="next Logo"
    src="https://seeklogo.com/images/C/contentful-logo-C395C545BF-seeklogo.com.png"
    style={
      Object {
        "height": "50px",
        "objectFit": "contain",
        "width": "50px",
      }
    }
  />
</OverlayTrigger>
`;

exports[`FooterImage Ensure FooterImage changes after mouse hover 3`] = `
<OverlayTrigger
  defaultShow={false}
  overlay={
    <Tooltip
      id="button-tooltip-2"
      placement="right"
    >
      This is a sample Footer Image
    </Tooltip>
  }
  placement="top"
  trigger={
    Array [
      "hover",
      "focus",
    ]
  }
>
  <img
    alt="next Logo"
    src="https://seeklogo.com/images/C/contentful-logo-C395C545BF-seeklogo.com.png"
    style={
      Object {
        "height": "50px",
        "objectFit": "contain",
        "width": "50px",
      }
    }
  />
</OverlayTrigger>
`;
